i18n/en.json

```diff
@@ -320,6 +320,26 @@
     "translation": "%s updated the channel header to: %s"
   },
   {
+    "id": "api.channel.post_update_channel_purpose_message_and_forget.join_leave.error",
```

Comment:
Sorry to be pedantic, but could you rename this error to replace the `join_leave.error` part with something more specific to this situation, e.g. `create_post.error`. You can also do the same change to the header error you tidied up above. Other than that, this PR looks fine to me.